kafka-rest/src/main/java/io/confluent/kafkarest/RestCustomRequestLog.java

```diff
@@ -0,0 +1,74 @@
+package io.confluent.kafkarest;
+
+import java.io.IOException;
+import org.eclipse.jetty.server.CustomRequestLog;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.RequestLog;
+import org.eclipse.jetty.server.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * RestCustomRequestLog implements Jetty's RequestLog interface. It offers the same log-format
+ * jetty's CustomRequestLog. Additionally it would append configured request-attributes (see
+ * requestAttributesToLog) to the end of the log.
+ */
+
+public class RestCustomRequestLog implements RequestLog {
+
+  private static final Logger log = LoggerFactory.getLogger(RestCustomRequestLog.class);
+
+  private RequestLog.Writer logWriter;
+  private String formatString;
+
+  private String[] requestAttributesToLog;
+
+  public RestCustomRequestLog(
+      RequestLog.Writer writer, String formatString, String[] requestAttributesToLog) {
+    this.logWriter = writer;
+    this.formatString = formatString;
+    this.requestAttributesToLog = requestAttributesToLog;
+  }
+
+  @Override
+  public void log(final Request request, final Response response) {
+    try {
+      StringBuilder logStringBuilder = new StringBuilder();
+      RequestLog.Writer jettyLogWriter = new StringBuilderLogWriter(logStringBuilder);
+      RequestLog jettyLog = new CustomRequestLog(jettyLogWriter, formatString);
```

Comment:
this approach is working for the purpose, however, I'll have a bit of concern about the performance, as for every log line, we will create a new `CustomRequestLog` and a new `StringBuilder` objects. While StringBuilder might be lightweight, `CustomRequestLog` is a big one.

What do you think?